src/build.rs

```diff
@@ -40,6 +49,12 @@ pub struct CheckoutBuilder<'cb> {
 /// of completed steps so far, and the final is the total number of steps.
 pub type Progress<'a> = FnMut(Option<&Path>, usize, usize) + 'a;
 
+/// Checkout notifications callback.
+///
+/// The first argument is the notification type, the next is the path for the
+/// the notification, followed by the baseline diff, target diff, and workdir diff.
+pub type Notify<'a> = FnMut(CheckoutNotificationType, Option<&Path>, DiffFile, DiffFile, DiffFile) -> c_int + 'a;
```

Comment:
It definitely shouldn't have been `c_int`, but I had left it as an integer before because the the docs say:

> Returning a non-zero value from this callback will cancel the checkout. The non-zero return value will be propagated back and returned by the git_checkout_... call.

So the number they returned could be retrieved from the result of the checkout function. Getting information out like that seems like a C-ism to me, as far as libgit2 is concerned all it cares about is whether it's non-zero or not, if it is then it cancels the checkout. Since the callback is FnMut, users could get information out through the environment captures, so I agree that it'll be simpler to return a bool.

So instead of returning 0 the callback will return false, which will read as "don't cancel the checkout", whereas true would have the opposite effect.